### PR TITLE
Added docs for the SessionValueResolver

### PR DESCRIPTION
**Pull request by linaori**
Explains the functionality added in symfony/symfony#21164

### CODE REVIEW

File: controller.rst [hunk 1]
```diff
@@ -344,18 +344,17 @@ Symfony provides a nice session object that you can use to store information
...
 
-To retrieve the session, call
-:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::getSession`
-method on the ``Request`` object. This method returns a
-:class:`Symfony\\Component\\HttpFoundation\\Session\\SessionInterface` with easy
-methods for storing and fetching things from the session::
 
-    use Symfony\Component\HttpFoundation\Request;
+.. versionadded:: 3.3
+    The ability to request a ``Session`` in actions was introduced in Symfony 3.
```
Comment (GuilhemN) [inline controller.rst]: `3.3`

**Comment by iltar** [comment]
Rebased and addressed the remark. @xabbuh I forgot to link to this PR in the feature PR, but now that it's merged, the `On hold` can be removed.

File: controller.rst [hunk 2]
```diff
@@ -348,18 +348,17 @@ Symfony provides a nice session object that you can use to store information
...
-methods for storing and fetching things from the session::
 
-    use Symfony\Component\HttpFoundation\Request;
+.. versionadded:: 3.3
+    The ability to request a ``Session`` in actions was introduced in Symfony 3.3.
 
-    public function indexAction(Request $request)
-    {
-        $session = $request->getSession();
+To retrieve the session, add the :class:`Symfony\\Component\\HttpFoundation\\Session\\Session`
```
Comment (HeahDude) [inline controller.rst]: What about:
"To retrieve the session, type hint a controller argument with :class:`Symfony\\Component\\HttpFoundation\\Session\\SessionInterface` or any implementation:"
?
Then keep the example below which is much more useful to get access to method not available through the interface.

**Comment by iltar** [comment]
I've addressed the comments and added a small example of how to use the `Session` instead of the `SessionInterface` for `getFlashBag`

**Review by HeahDude — APPROVED** [review]
👍

**Comment by HeahDude** [comment]
This one is not on hold anymore, https://github.com/symfony/symfony/pull/21164 is merged.

File: controller.rst [hunk 3]
```diff
@@ -372,6 +371,23 @@ methods for storing and fetching things from the session::
 
 Stored attributes remain in the session for the remainder of that user's session.
 
+.. tip::
+
+    Every ``SessionInterface`` implementation is supported. If you have your
+    own implementation, type-hint this in the arguments instead.
+
+As a developer, you might prefer not to extend the ``Controller``. To use the
```
Comment (wouterj) [inline controller.rst]: This paragraph should be moved to the end of the next section ("Flash Messages") and can maybe be put in a `.. sidebar::` or `.. tip::` directive instead.

File: controller/argument_value_resolver.rst [hunk 4]
```diff
@@ -18,6 +18,9 @@ functionality.
 Functionality Shipped with the HttpKernel
 -----------------------------------------
 
+.. versionadded:: 3.3
+    The ``SessionValueResolver`` was introduced in Symfony 3.3.
+
 Symfony ships with four value resolvers in the HttpKernel component:
```
Comment (wouterj) [inline controller/argument_value_resolver.rst]: four -> five

**Review by wouterj — APPROVED** [review]
Thanks! Left 2 minor comments that can be applied during the merge.

**Comment by xabbuh** [comment]
Thank you @iltar.

**Pull request closed without merging**